src/dataframerow/utils.jl

```diff
@@ -94,6 +94,59 @@ isequal_row(cols1::Tuple{Vararg{AbstractVector}}, r1::Int,
     isequal(cols1[1][r1], cols2[1][r2]) &&
         isequal_row(Base.tail(cols1), r1, Base.tail(cols2), r2)
 
+# Simple vector type for internal use which represents a virtual DataAPI.refpool
+# so that a vector of reals can be its own DataAPI.refarray:
+# this just allows telling the generic row_group_slots method for generic refpools
+# what is the minimum index and the number of (potential) groups
+# missing values are supported and represented used index max+1
+struct IntegerRefpool{T<:Union{Real, Missing}} <: AbstractVector{T}
+    min::Int
+    max::Int
+end
+
+Base.size(x::IntegerRefpool{T}) where {T} = (x.max - x.min + 1 + (T >: Missing),)
+Base.axes(x::IntegerRefpool{T}) where {T} = (x.min:(x.max + (T >: Missing)),)
```

Comment:
we should check then upon construction that `x.max + 1` does not overflow. The same for `x.max - x.min + 1 + (T >: Missing)`. Alternatively here we can perform `checked_add` here (it is slower than normal one, but it probably does not matter here)

Reply:
same with `cheked_sub`